File: wwld_backend/demo/src/main/java/com/gateway/service/ConceptService.java
package com.gateway.service;


import com.gateway.bussiness.businessImpl.ConceptBusinessImpl;
import com.gateway.dto.ConceptDTO;
import com.gateway.entity.Concept;
import com.gateway.request.ConceptRequest;
import com.gateway.response.ApiResult;
import com.gateway.response.ConceptResponse.ConceptResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.MediaType;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/concept")
public class ConceptService extends BaseFuntion{
    private static final Logger LOGGER = LoggerFactory.getLogger(ConceptService.class);

    @Autowired
    ConceptBusinessImpl conceptBusinessImpl;

    @RequestMapping(value = "findConcepts",method = RequestMethod.POST,produces = MediaType.APPLICATION_JSON_VALUE)
    @ResponseBody
    public ResponseEntity<ConceptResponse> findConcepts(@RequestBody ConceptRequest request){
        ConceptResponse response = new ConceptResponse();
        response.setBaseResponse(getBase(request));
        try {
            if (response.getResult().isOk()) {
                List<ConceptDTO> concepts = conceptBusinessImpl.findConcepts(request);
                response.setListConcepts(concepts);
            }
        } catch (Exception e) {
            LOGGER.error("Error while getting concepts ", e);
        }
        return ResponseEntity.ok(response);
    }

    @RequestMapping(value = "insert", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
    @ResponseBody
    public ResponseEntity<ConceptResponse> insert(@RequestBody ConceptRequest request) {
        ConceptResponse response = new ConceptResponse();
        response.setBaseResponse(getBase(request));
        try {
            if (response.getResult().isOk()) {
                Concept concept = conceptBusinessImpl.createConcept(request);
                response.setResult(ApiResult.Result.OK);
            } else {
                response.setResult(ApiResult.Result.FAILD);
            }
        } catch (Exception e) {
            LOGGER.error("Error while inserting concept", e);
        }
        return ResponseEntity.ok(response);
    };

    @RequestMapping(value = "update", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
    @ResponseBody
    public ResponseEntity<ConceptResponse> update(@RequestBody ConceptRequest request) {
        ConceptResponse response = new ConceptResponse();
        response.setBaseResponse(getBase(request));
        try {
            if (response.getResult().isOk()) {
                Concept concept = conceptBusinessImpl.updateConcept(request);
                response.setResult(ApiResult.Result.OK);
            } else {
                response.setResult(ApiResult.Result.FAILD);
            }
        } catch (Exception e) {
            LOGGER.error("Error while updating concept", e);
        }
        return ResponseEntity.ok(response);
    }

    @RequestMapping(value = "delete", method = RequestMethod.POST, produces = MediaType.APPLICATION_JSON_VALUE)
    @ResponseBody
    public ResponseEntity<ConceptResponse> delete(@RequestBody ConceptRequest request) {
        ConceptResponse response = new ConceptResponse();
        response.setBaseResponse(getBase(request));
        try {
            if (response.getResult().isOk()) {
                conceptBusinessImpl.deleteConcept(request);
                response.setResult(ApiResult.Result.OK);
            } else {
                response.setResult(ApiResult.Result.FAILD);
            }
        } catch (Exception e) {
            LOGGER.error("Error while deleting concept", e);
        }
        return ResponseEntity.ok(response);
    };


}

File: wwld-next-app/src/lib/services/conceptService.ts
import { ConceptPayload } from "../types/concept";
import { apiClient } from "../apiClient";


// tạo mới
export const addConcept = async (payload: ConceptPayload) => {
  try {
    const response = await apiClient.post("/api/concept/insert", payload);
    
    return response.data;
  } catch (error) {
    console.error("Error adding concept:", error);
    throw error;
  }
};

// Lấy danh sách
export const fetchConcepts = async () => {
  try {
    const request = {
      clientTime: new Date().toISOString()
    };
    const response = await apiClient.post("/api/concept/findConcepts", request);
    console.log("response : ", response.data.listConcepts);
    
    return response.data.listConcepts;
  } catch (error) {
    console.error("Error fetching concept :", error);
    throw error;
  }
};

// Sửa
export const updateConcept = async (payload: ConceptPayload) => {
  try {
    const response = await apiClient.post("/api/concept/update", payload);
    console.log("payload :",payload);
    
    return response.data;
  } catch (error) {
    console.error("Error updating concept:", error);
    throw error;
  }
};

// Xóa
export const deleteConcept = async (payload : ConceptPayload) => {
  try {
    
    const response = await apiClient.post("/api/concept/delete", { payload });
    return response.data;
  } catch (error) {
    console.error("Error deleting concept:", error);
    throw error;
  }
};

// get one by id 

export const fetchOneConceptById = async (payload: ConceptPayload) => {
  try {
    console.log("payload :",payload );
    
    const response = await apiClient.post("/api/concept/getOneById", payload);
    console.log("response : ", response.data.conceptDTO);
    
    return response.data.conceptDTO;
  } catch (error) {
    console.error("Error fetching concept :", error);
    throw error;
  }
};

export type { ConceptPayload };
